d3b_utils/aws_bucket_contents.py

```diff
@@ -0,0 +1,61 @@
+import boto3
+import re
+
+
+def fetch_aws_bucket_obj_info(
+    bucket_name, search_prefixes=None, drop_folders=False, drop_path=False
+):
+    """
+    List the objects of a bucket and the size of each object. Returns a
+    list of dicts.
+
+    :param bucket_name: The name of the bucket.
+    :type bucket_name: str
+    :param search_prefixes: path to the specific folder in the given
```

Comment:
It's a filtering prefix or iterable of prefixes, not a folder path. _If_ you don't want to support filtering with partial filenames (e.g. /foo returns /foo1, /foo2, etc), then add the '/' at the end yourself if it's not there and don't make the caller do it. 

I don't have a preference for whether this supports partial names or just full directory names, though it is neat to support both (just change "path to specific folder" in this first comment line and then change the trailing slash directive to only be if you exclusively want to match full directory names).

Reply:
ditto on thinking it's a neat idea. I like the idea of this being able to do both... I'll go towards doing that.